athletics.lic

```diff
@@ -0,0 +1,223 @@
+custom_require.call(%w(common common-travel))
+
+class Athletics
+  
+  include DRC
+  include DRCT
+  
+  def initialize
+    @did_play = false
+    @no_instrument = false
+    @settings = get_settings()
+    @athletics_options = get_data('athletics').athletics_options
+    @song_list = get_data('perform').perform_options
+ 
+    Flags.add('ct-song', 'you finish playing')
+    play_song?
+    
+    @start_skill=DRSkill.getxp('Athletics')
+    @end_skill=@start_skill+20
+    if(@end_skill>30)
+      @end_skill=30
+    end
+
+    if @settings.hometown == 'Crossing' || @settings.hometown == 'Riverhaven'
+      crossing_athletics
+    elsif @settings.hometown == 'Shard'
+      if DRSkill.getrank('Athletics') <= 250
+        shard_athletics
+      else
+        if DRSkill.getrank('Stealth') >= 250
+          climb_practice(19460,"wall",true)
+        else
+          hib_athletics
+        end
+      end
+    elsif @settings.hometown == 'Hibarnhvidar'
+      hib_athletics
+    elsif @settings.hometown == 'Ratha'
+      ratha_athletics
+    end
+  end
+  
+  def crossing_athletics
+    if DRSkill.getrank('Athletics') >= 400
+      return if @researching
+      climb_branch
+    else
+      while DRSkill.getxp('Athletics') < @end_skill
+        @athletics_options
+          .reject { |data| @settings.avoid_athletics_in_justice && data['justice'] }
+          .each do |data|
+            break unless climb?(data['room'], data['targets'])
```

Comment:
It feels odd that the athletics check against the target is done as part of the return value of `climb?()` if that method returned false I expect something to have gone wrong with the climb, not that we're full on exp. I'd rather have a helper method `done_climbing?` that checks exp and then we can 
```
climb(data...)
break if done_climbing?
```